### DISCUSSION TITLE
How to keep the fixed position of facet values in HierarchicalMenu?

### DISCUSSION THREAD

**Discussion by jhaankit** [algolia/instantsearch#5318] [Q&A]
How to keep the fixed position of facet values in HierarchicalMenu?
Sometimes it jumps down and hides from the displayed list of items.

I am referring to the link https://react-instantsearch.netlify.app/storybook/?path=/story/hierarchicalmenu--playground
Select `Audio > Cables & Chargers > MP3 Player Cables`
When I reach level 3 selection, it goes down and hides from the visible area.
The only option to see it now is to click the show more button, but this behavior is not user-friendly.

**Comment by Haroenv** [comment]
This indeed seems like a bug, the count changes when MP3 Player Cables is selected from 1570 (the count of Audio in general) to the number of the MP3 Cables

**Reply by jhaankit** [reply]
Ok. Then what is the next step to resolve this bug?

**Reply by mikedavis77** [reply]
@jhaankit I have raised this with the product team internally as a bug

**Reply by sarahdayan** [reply]
This seems to have been introduced in `6.12.0-alpha.1`, this is the first version at which I reproduce the bug. I believe this is caused by the new `facetOrdering` option which impacts how items are sorted.

If you're not using facet ordering for now, you can set `facetOrdering={false}` on the widget to solve the issue until we fix it.

Example: https://codesandbox.io/s/elastic-water-8u2gvi?file=/src/App.js

**Reply by FabienMotte** [reply]
The Algolia index we use in Storybook was configured by default to sort the facet values for the hierarchical menu by the `Number of hits (descending)` in the "Algolia Dashboard > Configuration > Facet display" section.
That's why when refining on the 3rd level, the count of the root level changes and since facet ordering is configured to sort the facet values by the number of hits, the "Audio" category is positioned last.

On the "Facet display" configuration section of the Dashboard, defining the option "Sort all other values by" to `Alphabetical order (ascending)` for the hierarchical menu facet solves the issue.

<img width="925" alt="image" src="https://user-images.githubusercontent.com/662153/173081862-448fd487-c792-4e98-8c32-55dad1564609.png">

Thank you for finding and reporting the issue!
